go/pkg/launchpad/utils.go

```diff
@@ -0,0 +1,153 @@
+package launchpad
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/TERITORI/teritori-dapp/go/pkg/launchpadpb"
+	"github.com/TERITORI/teritori-dapp/go/pkg/merkletree"
+	"github.com/cosmos/gogoproto/proto"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/pkg/errors"
+)
+
+type WhitelistAddress struct {
+	value string
+}
+
+func NewWhitelistAddress(addr string) *WhitelistAddress {
+	return &WhitelistAddress{value: addr}
+}
+
+func (addr *WhitelistAddress) toStr() string {
+	return fmt.Sprintf("%s", addr.value)
+}
+
+func (addr *WhitelistAddress) toBytes() []byte {
+	return []byte(addr.toStr())
+}
+
+func (addr *WhitelistAddress) GetID() string {
+	return addr.value
+}
+
+func (addr *WhitelistAddress) CalculateHash() ([]byte, error) {
+	res := crypto.Keccak256(addr.toBytes())
+	return res, nil
+}
+
+func (addr *WhitelistAddress) Equals(other merkletree.Content) (bool, error) {
+	otherVal := other.(*WhitelistAddress).toStr()
+	return otherVal == addr.toStr(), nil
+}
+
+func (addr *WhitelistAddress) ToJSONB() interface{} {
+	return map[string]interface{}{}
+}
+
+type Metadata struct {
+	launchpadpb.Metadata
+}
+
+func NewMetadataFromPb(data *launchpadpb.Metadata) *Metadata {
+	return &Metadata{
+		Metadata: launchpadpb.Metadata{
+			Image:                 data.Image,
+			ImageData:             data.ImageData,
+			ExternalUrl:           data.ExternalUrl,
+			Description:           data.Description,
+			Name:                  data.Name,
+			Attributes:            data.Attributes,
+			BackgroundColor:       data.BackgroundColor,
+			AnimationUrl:          data.AnimationUrl,
+			YoutubeUrl:            data.YoutubeUrl,
+			RoyaltyPercentage:     data.RoyaltyPercentage,
+			RoyaltyPaymentAddress: data.RoyaltyPaymentAddress,
+		},
+	}
+}
+
+func (m *Metadata) proto_encode() ([]byte, error) {
+	bytes, err := proto.Marshal(&m.Metadata)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to proto_encode metadata to proto")
+	}
+
+	return bytes, nil
+}
+
+// Converts the Metadata data into one string
+func (m *Metadata) GetID() string {

```

Comment:
What is this for :-s I don't really remember

Reply:
Can we replace this by an increased number id ?

Reply:
No, it's more complex than just making an ID: https://github.com/TERITORI/teritori-dapp/pull/1024/commits/6e1955c8d684981cd2e33ce3309fd6456a0b493c

See https://discord.com/channels/876922049279455274/1192401307559018569/1290091481830260797 and https://discord.com/channels/876922049279455274/1192401307559018569/1291461904698441791